analysis_engine/tracking/bat_tracker.py

```
@@ -0,0 +1,253 @@
+        "bat_face_angle_deg": 0.0,
+        "follow_through_ratio": 0.0,
+        "follow_through_label": "Unknown",
+        "sweet_spot_ratio": 0.72,
+    }
+    if len(tips) < 3 or fps <= 0:
+        return result
+
+    scale = (fps / max(1e-6, px_per_meter)) * 3.6
+    speeds: List[Tuple[int, float]] = []
+    arc = 0.0
+    for (i0, p0), (i1, p1) in zip(tips, tips[1:]):
```

Ruff (0.16.1)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)